### ISSUE TITLE
null error

### ISSUE THREAD

**Issue by OfficiallyBrown** [GraveofBears/GoodestBoy-Latest#8]
I have around 40 mods on my client, and a few less on my server but was able to narrow it down to this mod, but I was getting this infinitely spamming my console:

[Error : Unity Log] NullReferenceException: Object reference not set to an instance of an object Stack trace: FootStep.OnFoot (UnityEngine.Transform foot) (at <79989041713a40fba2973f1624473248>:0) FootStep.OnFoot () (at <79989041713a40fba2973f1624473248>:0) CharacterAnimEvent.FootStep (UnityEngine.AnimationEvent e) (at <79989041713a40fba2973f1624473248>:0)

everything was still working as expected. I could still tame dogs and use all the items. Given what the message said, the animations looked fine, but it didnt seem like it was making any noise when walking, not sure if its supposed to or not.

The game would run fine, but if I scrolled up in the console to check any other messages it would freeze and crash

**Comment by GraveofBears** [comment]
Is there some way to reproduce this?
I removed footstep audio when making the mod but perhaps there is an event on the animation to trigger foot step audio. I will check and remove it.

**Comment by GraveofBears** [comment]
I just updated, removed some missing animation on the controller that valheim is also missing on their wolf animator. I also changed all footstep effects and registered a variant of them. I do not think this will fix footstep effects, however it should prevent any null reference to them.

**Comment by OfficiallyBrown** [comment]
Sorry, I was at work. I installed the updated version on my client and server and was still getting the same error. I played around with it a bit, removing this mod and the only other one I have that adds creatures (OdinHorse), I would get the error anytime this mod was installed. Also, I didnt get that error on singleplayer, only when connecting to my server. I decided to try again with all my mods enabled (including this one), and I wasnt getting the error anymore. I restarted everything just to double check and I still wasn't getting the error, so I'm not really sure what exactly the problem was. 

Thanks for the fix though

Edit: Normally the error would start coming up as soon as I connected, apparently this time it just took a minute to start coming up, cuz I'm getting the same error again. If you would like to look at my modlist/logs I can post them. I can also help with any other troubleshooting you would like to try. Not sure if it matters, but I host my server on my own hardware (separate machine from my client) so getting into any files is no problem

**Comment by OfficiallyBrown** [comment]
If you use Thunderstore, this is my client modlist code 019093b2-062c-5712-5adc-446a8a28615e
<details>
  <summary>Mod List</summary>
  
PlantEverything v1.18.1 by Advize
PerfectPlacement v1.1.7 by Azumatt
AllTheBases v1.0.10 by Azumatt
ProfitablePieces v1.0.3 by Azumatt
AzuAreaRepair v1.1.4 by Azumatt
FirstPersonMode v1.3.6 by Azumatt
AzuWorkbenchTweaks v1.0.3 by Azumatt
AzuMiscPatches v1.2.2 by Azumatt
AzuWearNTearPatches v1.0.6 by Azumatt
AzuContainerSizes v1.0.3 by Azumatt
AzuExtendedPlayerInventory v1.4.3 by Azumatt
ImFRIENDLY_DAMMIT v1.1.7 by Azumatt
Official_BepInEx_ConfigurationManager v18.3.0 by Azumatt
Sleepover v1.1.3 by Azumatt
PetPantry v1.0.1 by Azumatt
FastLink v1.4.2 by Azumatt
CraftFromContainers v3.5.2 by Buttergeland
FloatingItems v1.0.0 by castix
BepInExPack_Valheim v5.4.2202 by denikson
Ballista_Reloader v0.217.25 by DFG
InfiniteTorches v0.0.2 by Frogger
Quick_Stack_Store_Sort_Trash_Restock v1.4.8 by Goldenrevolver
Server_devcommands v1.81.0 by JereKuusela
Item_Stand_All_Items v1.21.0 by JereKuusela
FascinatingCarryWeight v1.2.32 by kangretto
MultiUserChest v0.5.10 by MSchmoecker
VNEI v0.15.2 by MSchmoecker
OdinArchitect v1.3.0 by OdinPlus
OdinHorse v1.2.6 by OdinPlus
GoodestBoy v0.2.3 by OdinPlus
Basements v1.1.8 by OdinPlus
UsefulPaths v1.0.2 by RustyMods
CameraTweaks v1.2.0 by Searica
ComfortTweaks v3.3.1 by Smoothbrain
Sailing v1.1.7 by Smoothbrain
TargetPortal v1.1.20 by Smoothbrain
Jotunn v2.20.1 by ValheimModding
Venture_Multiplayer_Tweaks v0.11.2 by VentureValheim
Pet_Protection v0.2.0 by zebediah49
ValheimRAFT v2.2.1 by zolantris
</details>
The server mods are the same, minus FirstPersonMode, ImFRIENDLY_DAMMIT, BepInEx ConfigManager, FastLink, Ballista Reloader, FascinatingCarryWeight, VNEI, CameraTweaks, and Sailing

This is the servers BepInEx Log https://pastebin.com/7fqkLVku
And the Client https://pastebin.com/q8yyJgt3

I also get the error when this is the only mod installed on the server

**Comment by GraveofBears** [comment]
Unfortunately I wasnt able to get your profile to load, I was getting spammed with errors, so I tested with Goodestboy on my local client and server by itself. I do not seem to have any errors.
Are you able to generate some fresh logs? these appear to be from July 1st and the mod was updated on the 8th.

**Comment by OfficiallyBrown** [comment]
Yeah, I can do that in a couple hours when I get home. I thought I did do fresh logs, but I can redo it quick

Edit - I think the 7/1 date you were seeing was just the BepInEx version date. If you scroll down towards the bottom it says 7/8 but I did run some new ones. I also set the log levels to all, so particularly in the new client one, anything relevent is probably towards the bottom

These logs are with only GoodestBoy on the server, and GoodestBoy and FastLink on the client
Server - https://pastebin.com/8EDu6A53
Client - https://pastebin.com/LXCv628f
